### DISCUSSION TITLE
Open Search configuration

### DISCUSSION THREAD

**Discussion by Mezzenine** [Opster/opensearch-k8s-operator#179] [Q&A] [answered]
Hi,

In the OpenSearch documentation, it is mentioned under important setting, to set the vm.max_map_count=262144. In a docker environment, this must be set on the host machine. In Kubernetes, I assume it must be set on the node. 
When you are running an AKS, the underlying nodes are a VM scale-set and you don't want to configure this on the node level.

Is this setting obsolete with the operator or do I have other possibilities to apply this setting?

**Comment by swoehrl-mw** [comment, marked as answer]
Hi @Mezzenine. The cluster CRD has the field `spec.general.setVMMaxMapCount`, if it is set to `true` then the operator will add an init container to the opensearch pods to set `vm.max_map_count` as the documentation describes. 
Hope this helps.

**Comment by mrvdcsg** [comment]
What @swoehrl-mw said is absolutely the right path. However, if you run into permissions issues of init containers having access to alter scale set host attributes like vm.max_map_count, you can automate that change as part of the scale set itself. I have done this in my terraform build and apply it to every scale set that will run OS. 

        sysctl_config = {
          vm_max_map_count = var.vm_max_map_count
        }
You can find details on how to customize your node configuration for Azure AKS here: https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration

**Comment by kedagema** [comment]
I am building a OpenSearch cluster in GKE Autopilot mode, I have set VMMaxmapcount "true" and build custom base image for init-helper and specified in cluster spec. I am seeing a permission issue for creating resources using init-helper container as it requires privileged access.

My google admin team is suggesting to deploy using daemonset to overcome this issue. I see OpenSearch operator is using statefulset for deployment. May I know which part of the code I need to change to do daemonset deployment.

Or is there an workaround to configure init-container sysctl commands in the spec. let me know if you need any further information. Please help. Thanks
